src/random.cpp

```diff
@@ -671,9 +671,11 @@ void MakeRandDeterministicDANGEROUS(const uint256& seed) noexcept
 {
     GetRNGState().MakeDeterministic(seed);
 }
+std::atomic<bool> g_used_g_prng{false}; // Only accessed from tests
 
 void GetRandBytes(Span<unsigned char> bytes) noexcept
 {
+    g_used_g_prng = true;
```

Comment:
This is about excluding test-only code from production (currently fuzz-only since *check_globals.h/cpp* is in *fuzz/util/*). Not sure why that would be undesirable. Even though the conditional adds verbosity, it adds context and decreases need for comments.

Might be worth introducing a `TESTS_ENABLED` `#define`... but I'm guessing that has been discussed before?